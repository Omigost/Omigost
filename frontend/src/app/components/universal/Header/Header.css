@import "../../../css/Theme.less";

.Header {
  font-family: @PrimaryFont;
  background-color: black;
  color: white;
  height: 100vh;
  max-height: 100vh;
  width: 100vw;
  overflow-x: hidden;
}

.HeaderMenuWrapper {
  position: absolute;
  width: 100%;
  z-index: 9999999;
}

.ExpandableMenuWrapper {
  position: absolute;
  display: none;
  top: 0;
  left: 0;
  z-index: 99999999;
}

.FullHeader {
  font-family: @PrimaryFont;
  background-color: black;
  color: white;
  height: 100vh;
  max-height: 100vh;
  width: 100vw;
  overflow-x: hidden;
}

.NotLimitedImageHeader {
    height: unset !important;
    max-height: unset !important;
    overflow-y: visible !important;
}

.HeaderElementWrapper {
  position: absolute;
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
}

.HeaderElementWrapperCentered {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  width: 100%;
  height: 100%;
  height: @HeaderHeight;
  padding-top: @HeaderHeight / 4;
}

.BackgroundImageWrapperFilter {
  overflow: hidden;
  height: 100%;
  width: 100%;
  background: rgba(0, 0, 0, 0.478431);
}

.BackgroundImage {
  height: 100%;
  width: 100%;
  background: @BaseGradient;
}

.BackgroundImageWrapper {
  overflow: hidden;
  height: 100%;
  width: 100%;
}

.BackgroundImage {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.BackgroundImageFull {
  min-height: 100vw;
}

.ForegroundWrapper {
  position: absolute;
  overflow: hidden;
  max-width: 100vw;
}

.Foreground {
  height: 131vh;
  position: relative;
  top: -31vh;
  left: -3vw;
}

@media only screen and (max-width: 600px) {
  .Foreground {
    height: 133vh;
    top: 28vh;
  }
}

@media only screen and (max-width: 435px) {
  .ExpandableMenuWrapper {
    display: block;
  }
}

@media only screen and (max-width: 375px) {
  .Foreground {
    height: 111vh;
    top: 51vh;
  }
}

@media only screen and (max-width: 215px) {
  .Foreground {
    height: 101vh;
    top: 45vh;
    left: -58vw;
  }
}
