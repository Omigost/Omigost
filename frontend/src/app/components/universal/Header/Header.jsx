import React from 'react'
import * as styles from './Header.css'

import HeaderBar from 'components/HeaderBar/HeaderBar.jsx'
import MainPanel from 'components/MainPanel/MainPanel.jsx'
import HeaderMenu from 'components/HeaderMenu/HeaderMenu.jsx'
import TopExpandableMenu from 'components/TopExpandableMenu/TopExpandableMenu.jsx'


class Header extends React.Component {
  render() {
    let {
      headerElement,
      fullScreen,
      centerHeaderElement,
      doNotLimitBackgroundImage,
      fullBackgroundImage
    } = this.props;

    if(!headerElement) {
      headerElement = (
        <MainPanel />
      );
    }

    return (
      <div
        styleName={
          (fullScreen)?('FullHeader'):('Header')
        }
        className={
          (doNotLimitBackgroundImage)?(styles.NotLimitedImageHeader):(null)
        }
       >
        <HeaderBar />
        <div styleName='HeaderMenuWrapper'>
            <HeaderMenu />
        </div>
        <HeaderMenu />
        <div styleName='BackgroundImageWrapper'>
          <div
            styleName='BackgroundImage'
            className={
              (fullBackgroundImage)?(styles.BackgroundImageFull):('')
            }
          />
        </div>
        <div
          styleName={
            (centerHeaderElement)?(
              'HeaderElementWrapperCentered'
            ):('HeaderElementWrapper')
          }
        >
          {headerElement}
        </div>
        <div styleName='ExpandableMenuWrapper'>
          <TopExpandableMenu coverScreen={true}/>
        </div>
      </div>
    )
  }
}

//<!--<div styleName='BackgroundImageWrapperFilter'></div>-->

export default Header
