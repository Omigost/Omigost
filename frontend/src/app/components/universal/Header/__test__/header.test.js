import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

import Header from '../Header.jsx';

test('Snapshot test of Header', () => {
    const component = shallow(
        <Header />
    );

    expect(toJson(component, {
        mode: 'shallow'
    })).toMatchSnapshot();
});