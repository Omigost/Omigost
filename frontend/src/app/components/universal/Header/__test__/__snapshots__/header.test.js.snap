// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Header 1`] = `
<div
  styleName="Header"
>
  <HeaderBar />
  <div
    styleName="HeaderMenuWrapper"
  >
    <PropsContentProvider />
  </div>
  <PropsContentProvider />
  <div
    styleName="BackgroundImageWrapper"
  >
    <img
      src="img/backgrounds/home2g.png"
      styleName="BackgroundImage"
    />
  </div>
  <div
    styleName="HeaderElementWrapper"
  >
    <PropsContentProvider />
  </div>
  <div
    styleName="ExpandableMenuWrapper"
  >
    <TopExpandableMenu
      coverScreen={true}
    />
  </div>
</div>
`;
